app/models/donation.rb

```diff
@@ -0,0 +1,92 @@
+class Donation
+  include ActiveModel::Model
+
+  attr_accessor :email, :employer, :occupation, :stripe_token, :recurring,
+    :utm_source, :utm_medium, :utm_campaign, :source_url, :amount_in_cents
+
+  validates :email, presence: true,
+    email_format: { message: 'is not a valid email' }
+  validates :employer, presence: true
+  validates :occupation, presence: true
+  validates :stripe_token, presence: true
+  validates :amount_in_cents, presence: true,
+    numericality: { greater_than: 0, only_integer: true }
+
+  def process!
+    if valid?
+      process_payment
+      record_donation
+      create_donate_action
+    else
+      false
+    end
+
```

Comment:
the `end` comes after `rescue Stripe::CardError`. Rescuing Stripe errors should probably go in `process_payment` method, but this was easier for now. I still think Stripe stuff should be extracted into new class in which case error handling would go with it. (Maybe that class would also have a `cancel_subscription` method.)